app/filters.js

```diff
@@ -0,0 +1,46 @@
+const dateFilter = require('nunjucks-date');
+
+module.exports = function (env) {
+  /**
+   * Instantiate object used to store the methods registered as a
+   * 'filter' (of the same name) within nunjucks. You can override
+   * gov.uk core filters by creating filter methods of the same name.
+   * @type {Object}
+   */
+  let filters = {}
+
+  filters.date = dateFilter;
+
+  function mojDateFilter(timestamp, type) {
+
+    // default datetime to timestamp
+    let datetime = timestamp;
+
+    switch(type) {
+      case "longdatetime":
+        datetime = dateFilter(timestamp, 'D MMMM gggg') + " at " + dateFilter(timestamp, 'h:mm A');
+        break;
+      case "shortdatetime":
+        datetime = dateFilter(timestamp, 'D MMM gggg') + " at " + dateFilter(timestamp, 'h:mm A');
+        break;
+      case "longdate":
+        datetime = dateFilter(timestamp, 'D MMMM gggg');
+        break;
+      case "shortdate":
+        datetime = dateFilter(timestamp, 'D MMM gggg');
+        break;
+      case "time":
+        datetime = dateFilter(timestamp, 'h:mm A');
```

Comment:
This isn't quite right as it will always put minutes in. So 9am should be 9am (not 9:00am)

Reply:
In line with `date` and `shortdate`, `datetime` and `shortdatetime`, I will introduce `shorttime` (9 AM) alongside `time` (9:00 AM).

What is the guidance around "am" versus "AM"? Aside: Grammarly always corrects me to uppercase.

Reply:
Is there a need for a `longdatetime`, for example: `Tuesday, 21 January 2019 at 2:15 PM`?

Reply:
The GOV.UK style recommends always 9am and not 9:00am. So we shouldn't provide that option I don't think. It's also lowercase “am”.

Reply:
For case working systems, it's necessary to be more accurate with time than hours. The provision of `time` and `shorttime` gives this flexibility and follows `date` and `datetime` nomenclature.